File: native/ios/AllStakRNModule.m
// RCTBridgeModule bridging AllStakCrashHandler to JS.
//
// SCAFFOLDED: requires React Native iOS project with CocoaPods autolinking
// to verify end-to-end.

#import <React/RCTBridgeModule.h>
#import "AllStakCrashHandler.h"

@interface AllStakRNModule : NSObject <RCTBridgeModule>
@end

@implementation AllStakRNModule

RCT_EXPORT_MODULE(AllStakNative);

RCT_EXPORT_METHOD(install:(NSString *)release
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject) {
    [AllStakCrashHandler installWithRelease:release];
    resolve(@YES);
}

RCT_EXPORT_METHOD(drainPendingCrash:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject) {
    NSString *json = [AllStakCrashHandler drainPendingCrash];
    resolve(json ?: [NSNull null]);
}

@end

File: native/android/src/main/java/io/allstak/rn/AllStakRNModule.java
package io.allstak.rn;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.Promise;

/**
 * React Native module bridging {@link AllStakCrashHandler}.
 *
 *  JS side:
 *     NativeModules.AllStakNative.drainPendingCrash().then(json => ...)
 *
 * SCAFFOLDED — compiles against React Native; requires a bare RN app with
 * autolinking to verify end-to-end.
 */
public class AllStakRNModule extends ReactContextBaseJavaModule {
    public AllStakRNModule(ReactApplicationContext ctx) { super(ctx); }

    @NonNull
    @Override
    public String getName() { return "AllStakNative"; }

    @ReactMethod
    public void install(String release, Promise promise) {
        try {
            AllStakCrashHandler.install(getReactApplicationContext(), release);
            promise.resolve(true);
        } catch (Throwable t) {
            promise.reject("install-failed", t);
        }
    }

    @ReactMethod
    public void drainPendingCrash(Promise promise) {
        try {
            String json = AllStakCrashHandler.drainPendingCrash(getReactApplicationContext());
            promise.resolve(json);
        } catch (Throwable t) {
            promise.reject("drain-failed", t);
        }
    }
}
